docs/src/understand/team-feature-settings.md

```diff
@@ -72,6 +72,70 @@ galley:
             lockStatus: locked
 ```
 
+## Self-deleting messages
+
+Self-deleting messages is a feature that makes all sent messages delete after a given period of time the message has been read. By default, this feature is unlocked and not enabled for all the teams. The feature can be toggled on/off through Team Settings app. Team settings app has presets that can set up message expiration up to a month. Other time frames can be set either through a backend-wide global setting, or manually through API.
+
+Example of backend-wide global setting:
+
+```yaml
+galley:
+  # ...
+  config:
+    # ...
+    settings:
+      # ...
+      featureFlags:
+        # ...
+        selfDeletingMessages:
+          defaults:
+            config:
+              enforcedTimeoutSeconds: 0 # expressed in seconds
+            lockStatus: unlocked # locked | unlocked
+            status: enabled # enabled | disabled
+```
+
+Locking a feature will prevent team administrators from changing the setting!
+Backend-wide settings are not applied retroactively! For teams made before a global setting has been set, feature will have to be set manually through a combination of API calls.
+
+### Self-deleting messages custom time frame with API calls
+
+As a on-prem backend administrator, if you find yourself in need of a time-frame not provided by the Team Setting app. You can make an API call to `galley` to set it manually.
+
+First, port-forward the galley service:
+
+```bash
+d kubectl port-forward svc/galley 8080:8080
+```
+
+And make a curl request like in example below, set the appropriate <teamID> and <TeamAdminUserId> (these can be fetched from cassandra with `cqlsh` if you are not a direct owner of the team or the admin user)
+
+```bash
+curl -X GET 'http://localhost:8080/teams/<teamID>/features/selfDeletingMessages' -H 'accept: application/json;charset=utf-8' -H 'Content-Type: application/json' -H 'Z-User: <TeamAdminUserId>'   --data-raw '{ "config": { "enforcedTimeoutSeconds": 6000}, "status": "enabled" }'
+```
+
+For locking/unlocking features for a team, use of `backoffice` will be required.
```

Comment:
This can also be done via the internal API. Not sure if backoffice is available on on-prem installations.